test/Microsoft.AspNet.TestHost.Tests/TestServerTests.cs

```diff
@@ -38,6 +38,23 @@ public void ThrowsIfNoApplicationEnvironmentIsRegisteredWithTheProvider()
         }
 
         [Fact]
+        public async Task CanAccessHttpContext()
+        {
+            var services = new ServiceCollection().BuildServiceProvider();
+            TestServer server = TestServer.Create(app =>
+            {
+                app.Run(context =>
+                {
+                    var accessor = app.ApplicationServices.GetRequiredService<IHttpContextAccessor>();
```

Comment:
A better test would to be to do this to a singleton service and called from here